src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java

```diff
@@ -144,4 +155,115 @@ public String getAddForeignKeyConstraintString(
   }
+
+  /* Lock acquisition functions */
+
+  @Override
+  public boolean supportsLockTimeouts() {
+    return false;
+  }
+
+  @Override
+  public LockingStrategy getLockingStrategy(Lockable lockable, LockMode lockMode) {
+    return LOCKING_STRATEGY;
+  }
+
+  @Override
+  public String getForUpdateString(LockOptions lockOptions) {
+    return "";
+  }
+
+  @Override
+  public String getForUpdateString() {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getForUpdateString(String aliases) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getForUpdateString(String aliases, LockOptions lockOptions) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getWriteLockString(int timeout) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getWriteLockString(String aliases, int timeout) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getReadLockString(int timeout) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getReadLockString(String aliases, int timeout) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public boolean supportsOuterJoinForUpdate() {
+    return false;
+  }
+
+  @Override
+  public String getForUpdateNowaitString() {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getForUpdateNowaitString(String aliases) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+
+  @Override
+  public String getForUpdateSkipLockedString() {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getForUpdateSkipLockedString(String aliases) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String applyLocksToSql(String sql, LockOptions aliasedLockOptions,
+      Map<String, String[]> keyColumnNames) {
+    return sql;
+  }
+
+  /**
+   * A locking strategy for the Cloud Spanner dialect that does nothing. Cloud Spanner does not
+   * support locking.
```

Comment:
How does transaction-level locking fit into this?
https://cloud.google.com/spanner/docs/transactions

Reply:
The locking here is at the query-level (part of SELECT queries) while the transaction locking setting is at the session level (you use a session to create many queries). The transaction locking and such isn't in the dialect from my examination and is handled by the driver implementation instead.